UI/window-basic-main-profiles.cpp

```diff
@@ -985,3 +989,178 @@ void OBSBasic::CheckForSimpleModeX264Fallback()
 		activeConfiguration.SaveSafe("tmp");
 	}
 }
+
+/* These may also be useful for QSV/AMF tests so keep them here. */
+static auto args_deleter = [](os_process_args_t *args) {
+	os_process_args_destroy(args);
+};
+using OsProcessArgs =
+	std::unique_ptr<os_process_args_t, decltype(args_deleter)>;
+
+static std::optional<const std::string> CheckNVENCVersion()
+{
+	const std::vector<std::pair<std::string_view, std::string_view>>
+		nvencFailureReasons = {
+			{"nvml_lib", "Failed loading NVML library"},
+			// Some erorr messages are a prefix + numeric error code
+			{"nvml_init_", "Failed initializing NVML"},
+			{"nvenc_lib", "Failed loading NVENC library"},
+			{"nvenc_init_", "Failed initializing NVENC"},
+			{"cuda_lib", "Failed loading CUDA library"},
+			{"cuda_init_", "Failed initializing CUDA"},
+			{"no_cuda_version", "Cuda version empty"},
+			{"no_devices", "No NVIDIA GPUs found"},
+			{"no_nvenc_version", "NVENC version empty"},
+			{"outdated_driver", "Outdated driver"},
+			{"no_supported_devices",
+			 "No NVIDIA devices with NVENC support found"},
+			{"session_limit", "Encoder session limit exceeded"},
+
+		};
+
+#ifdef _WIN32
+	BPtr test_exe = os_get_executable_path_ptr("obs-nvenc-test.exe");
+#else
+	BPtr test_exe = os_get_executable_path_ptr("obs-nvenc-test");
+#endif
+
+	OsProcessArgs args{os_process_args_create(test_exe), args_deleter};
+
+	os_process_pipe_t *pp = os_process_pipe_create2(args.get(), "r");
+	if (!pp)
+		return "NVENC check start failure";
+
+	std::string caps_result;
+	for (;;) {
+		char data[2048];
+		size_t len =
+			os_process_pipe_read(pp, (uint8_t *)data, sizeof(data));
+		if (!len)
+			break;
+
+		caps_result.append(data, len);
+	}
+
+	int exit_code = os_process_pipe_destroy(pp);
+
+	if (caps_result.empty()) {
+		return "NVENC check process failure, exit code: " +
+		       std::to_string(exit_code);
+	}
+
+	auto config = ConfigFile();
+	if (config.OpenString(caps_result.c_str()) != CONFIG_SUCCESS) {
+		return "NVENC check process output read failure, exit code: " +
+		       std::to_string(exit_code);
+	}
+
+	/* Read devices and attempt to find reason for failure */
+	auto numDevices = config_get_uint(config, "general", "cuda_devices");
+	if (!numDevices)
+		return "No NVIDIA GPUs found";
+
+	/* Check device generation (architecture) */
+	for (uint64_t i = 0; i < numDevices; i++) {
+		std::string section = "device." + std::to_string(i);
+
+		if (!config_has_user_value(config, section.c_str(), "name"))
+			continue;
+
+		const std::string_view name =
+			config_get_string(config, section.c_str(), "name");
+		const std::string_view arch = config_get_string(
+			config, section.c_str(), "architecture_name");
+
+		if (arch == "Kepler") {
+			std::string ret =
+				"NVIDIA Kepler (GTX 600/700 series) GPUs are no longer supported. Your GPU: ";
+			ret += name;
+
+			return ret;
```

Comment:
We should probably tell the user what the last OBS version is that supports Kepler (preferably with a link).

Reply:
The end goal was to link to a KB page in the message box, perhaps we could add a breadcrumb (e.g. `#kepler`) to point at the correct section?